pkg/resources/helpers.go

```diff
@@ -130,6 +130,35 @@ func GetPropertyAsPointer[T any](d *schema.ResourceData, property string) *T {
 	return &typedValue
 }
 
+func GetFirstNestedObjectByKey[T any](d *schema.ResourceData, propertyKey string, nestedValueKey string) *T {
```

Comment:
Described and I think it should be because it wouldn't be possible to test in `helpers_test.go`.

Reply:
testability shouldn't be the reason to make something exported. It would be testable as unexported method if helpers_test package was `resources` and not `resources_test`. I don't know if there was any deeper reason to have black-box tests for the helpers (maybe there were plans to extract them later?).

Let's leave it exported for now. (close after reading)